src/DeviceInterface.cpp

```diff
@@ -46,6 +47,32 @@ bool lookup_runtime_routine(const char *name, const Target &target,
 
 }
 
+namespace Halide {
+halide_device_interface_t *get_device_interface_for_target(const Target &t) {
```

Comment:
The comment claims that you have to pass it a target with exactly one set. If you set multiple it uses the same ordering preference as the gpu_tile, etc. Not sure if this is the best behavior, but it's important that:

    foo.copy_to_device();
    bar.gpu_tile(x, y, 8, 8);

does the right thing. Maybe it should take a DeviceAPI instead?